wasm/main.py

```diff
@@ -609,31 +639,34 @@ def spec_validate_start(C, start):
 # 3.4.8 EXPORTS
 
 
-def spec_validate_export(C, export):
-    return spec_validate_exportdesc(C, export["desc"])
+TExportValue = Union[FuncType, TableType, MemoryType, GlobalType]
 
 
-def spec_validate_exportdesc(C, exportdesc):
-    x = exportdesc[1]
-    if exportdesc[0] == "func":
-        if len(C["funcs"]) <= x:
+def spec_validate_export(C: Context, export: Export) -> TExportValue:
+    return spec_validate_exportdesc(C, export.desc)
+
+
+def spec_validate_exportdesc(C: Context,
+                             desc: ExportDesc) -> TExportValue:
+    if isinstance(desc, FuncIdx):
+        if len(C["funcs"]) <= desc:
```

Comment:
I don't follow what `desc` is supposed to mean here. How about something like `export_index`? (Same for the type name)

Reply:
I went with `index` which I'm very meh about but I see this as something that we can reasonably easily evolve the name as we find better ones.  Index for Indices which is what the value is, an index into the declared functions/tables/memory/globals.